runtime/JavaScript/src/antlr4/RuleContext.js

```diff
@@ -30,7 +30,6 @@ var INVALID_INTERVAL = require('./tree/Tree').INVALID_INTERVAL;
 var INVALID_ALT_NUMBER = require('./atn/ATN').INVALID_ALT_NUMBER;
```

Comment:
I think this should be: ...).ATN.INVALID_ALT_NUMBER;

Reply:
this is fine as it is because I changed the export of ATN.js to be `module.exports = ATN;` which means that the object itself is exported. This is precisely the equivalent of export default.

Reply:
so it's fine now? I thought you said it was undefined?

Reply:
sorry for the misunderstanding. The situation is as follows:
- the ATN module is exported directly so its members are directly available as shown above.
- the cyclic dependency prevents the ATN module from being correctly imported, therefore node just returns an empty object `{}` which results in `INVALID_ALT_NUMBER` being `undefined`. However, that has nothing to do with the way the object is exposed but rather with the cyclic dependency itself.